Add documentation for Code Security configuration file (#35307)
DataDog/documentation: content/en/security/code_security/static_analysis/setup/_index.md

* Repository-level configuration (Datadog)
* Repository-level configuration (Repo File)

All three locations use the same YAML format for configuration. These configurations are merged **in order** (see [how configurations interact](#how-org-level-repo-level-and-file-level-settings-interact) below).

#### Default rulesets

By default, Datadog's default rulesets are enabled for your repository's programming language(s) (`use-default-rulesets` is `true`). You can customize which rulesets are active:

- To **add** rulesets beyond the defaults, list them under `use-rulesets`.
- To **disable** specific rulesets, list them under `ignore-rulesets`.

> - To **disable all defaults** and choose your own, set `use-default-rulesets: false` and list the desired rulesets under `use-rulesets`.

Reviewer: ```suggestion
- **Add rulesets**: List them under `use-rulesets`
- **Disable specific rulesets**: List them under `ignore-rulesets`
- **Disable all default rulesets**: Set `use-default-rulesets: false`, then listing the desired rulesets under `use-rulesets`
```

For the full list of default rulesets, see [Static Code Analysis (SAST) Rules][6].

> #### How org-level, repo-level, and file-level settings interact

Reviewer: ```suggestion
#### Configuration precedence and merging
```

> All configuration levels use the same YAML format. These configurations are merged in order, with the following precedence (highest to lowest):

Reviewer: This felt slightly confusing to me. Thoughts on this? 

```suggestion
Configurations are merged in the following order, where each level overrides the one before it:
```

        max-class-lines:
          arguments:
            # Set different thresholds for the max-class-lines rule in different subtrees
            max-lines:
              # Set the rule's threshold to 200 lines by default (root path of the repo)
              /: 200
              # Set the rule's threshold to 100 lines in src/main/backend
              src/main/backend: 100
          # Override this rule's severity with different values in different subtrees
          severity:
            # Set the rule's severity to NOTICE by default
            /: NOTICE
            # Set the rule's severity to NONE in tests/
            tests: NONE
    python-django:
      # Only apply the python-django ruleset to the following paths
      only-paths:
        - "src/main/backend"
        - "src/main/django"
      # Do not apply the python-django ruleset in files matching the following pattern
      ignore-paths:
        - "src/main/backend/util/*.py"
  global-config:
    # Only analyze source files
    only-paths:
      - "src/main"
      - "src/tests"
      - "**/*.py"
    # Do not analyze third-party files
    ignore-paths:
      - "lib/third_party"
```

## Legacy configuration

> Datadog Static Code Analysis (SAST) configuration used to use a different file (``static-analysis.datadog.yml``) and a different schema. This schema has been deprecated and will not receive new updates, though it remains [documented][25] in the `datadog-static-analyzer` repository. If present, a ``code-security.datadog.yaml`` will always take precedence over ``static-analysis.datadog.yml``.

Reviewer: ```suggestion

Datadog Static Code Analysis (SAST) previously used a different configuration file (`static-analysis.datadog.yml`) and schema. This schema is deprecated and does not receive new updates, but it remains [documented][25] in the `datadog-static-analyzer` repository.

If both files are present, `code-security.datadog.yaml` takes precedence over `static-analysis.datadog.yml`.
```
